odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,254 @@
+    NIFTYREC_AVAILABLE = True
+except ImportError:
+    NIFTYREC_AVAILABLE = False
+
+import numpy as np
+
+
+from odl.operator import Operator
+from odl import uniform_discr_frompartition
+
+__all__ = ('SpectProject', 'SpectBackproject')
+_SUPPORTED_IMPL = ('niftyrec', 'niftyrec_gpu')
+
+
+class SpectProject(Operator):
+    """Discrete SPECT projector"""
+    def __init__(self, dom, geometry, attenuation=None, psf=None,
+                 impl='niftyrec'):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        dom : `DiscreteLp`
+            Discretized space, a 3 dimensional domain of the forward projector
+        geometry : geometry from :class:`~odl.tomo.geometry.spect`
+            The geometry of the transform. An ODL geometry instance
+            that contains information about the operator domain
+        attenuation : `element-like` structure of the shape
+            `dom.shape`, optional
+            Linear attenuation map for SPECT.
+        psf : `element-like` structure, optional
+            Point Spread Functions for the correction of the
+            collimator-detector response
+        impl={'niftyrec', 'niftyrec_gpu'}, optional
+            Implementation back-end for the transform. Supported back-ends:
+
+            'niftyrec' : `NiftyRec <http://niftyrec.scienceontheweb.net/\
+wordpress/>`_ tomography toolbox using the `niftypy <https://github.com/\
+spedemon/niftypy>`_ Python wrapper and CPU
+
+            'niftyrec_gpu' : `NiftyRec <http://niftyrec.scienceontheweb.net/\
+wordpress/>`_ tomography toolbox using the `NiftyPy <https://github.com/\
+spedemon/niftypy>`_ Python wrapper and GPU
+        """
+        impl, impl_in = str(impl).lower(), impl
+        if impl not in _SUPPORTED_IMPL:
+            raise ValueError('implementation {!r} not supported.'
+                             ''.format(impl_in))
+
+        if impl.startswith('niftyrec'):
+            if not NIFTYREC_AVAILABLE:
+                raise ValueError('NiftyRec back-end not available.')
+            if impl == 'niftyrec_gpu' and dom.shape != (128, 128, 128):
+                raise NotImplementedError('NiftyRec GPU back-end only '
+                                          'supports volume size '
+                                          '(128, 128, 128), got {}'
+                                          ''.format(dom.shape))
+            if impl == 'niftyrec_gpu':
+                self._use_gpu = True
+            elif impl == 'niftyrec':
+                self._use_gpu = False
+
+        self._geometry = geometry
+        det_nx, det_ny = geometry.det_grid.shape
+        startAngle = geometry.angles[0]
+        stopAngle = geometry.angles[-1]
+        N_proj = geometry.motion_grid.size
+
+        self._cameras = np.linspace(startAngle, stopAngle, N_proj,
+                                    dtype=np.float32).reshape((N_proj, 1))
+        if psf is not None:
+            self._psf = np.asarray(psf)
```

Comment:
@niinimaki and I discussed this. Usually the PSF is given as an array of small size and cannot simply be made an element of the domain. And I assume (without knowing better) that blowing it up to full size would bog down the code significantly. Would be worth checking, though. This is the obvious fix, delegating the PSF handling to Nifty.

If there is no difference in speed, we could simply add zeros to the kernel and go on. Otherwise, I'd add a TODO here, something like
```
# TODO: Create a smaller space for the PSF if necessary and convert to an element.
```